### DISCUSSION TITLE
Print and upload to onedrive

### DISCUSSION THREAD

**Discussion by FerreiraCode** [cypress-io/cypress#28242] [Questions and Help]
I'm starting my studies in Cypress and among the challenges I have to face is taking a screenshot of the screen when the tests give an error and sending this printout to a folder on a One Drive, could anyone help me? how can I do this?

**Comment by portyu9** [comment]
Hi @FerreiraCode 

You do not need to manually take a screenshot when a test fails during `cypress run` — Cypress already does that by default.

Failure screenshots are normally written under:

```text
cypress/screenshots
```

If OneDrive is synced or mounted as a local folder on the machine running Cypress, you can copy each screenshot there from the Node-side `after:screenshot` event:

```js
const { defineConfig } = require('cypress')
const fs = require('fs/promises')
const path = require('path')

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      on('after:screenshot', async (details) => {
        const oneDriveDir = process.env.ONEDRIVE_SCREENSHOT_DIR

        if (!oneDriveDir) {
          return
        }

        await fs.mkdir(oneDriveDir, { recursive: true })

        const destination = path.join(
          oneDriveDir,
          path.basename(details.path)
        )

        await fs.copyFile(details.path, destination)
      })
    },
  },
})
```

Set `ONEDRIVE_SCREENSHOT_DIR` to the path of the locally synced OneDrive folder.

If Cypress is running in CI and OneDrive is **not** mounted/synced there, uploading to OneDrive is outside Cypress itself. In that case, upload the generated files using Microsoft Graph or a CI step after the tests finish.

So the split of responsibilities is:

- **Cypress:** capture the failure screenshot.
- **Node/CI/OneDrive integration:** copy or upload the resulting file.

Please mark this answer as accepted if it helped, thank you.
